test/Libraries/Microsoft.Extensions.AI.Abstractions.Tests/Utilities/AIJsonUtilitiesTests.cs

```diff
@@ -160,9 +162,36 @@ public enum MyEnumValue
     }
 
     [Fact]
-    public static void ResolveJsonSchema_CanBeBoolean()
+    public static void CreateJsonSchema_CanBeBoolean()
     {
         JsonElement schema = AIJsonUtilities.CreateJsonSchema(typeof(object));
         Assert.Equal(JsonValueKind.True, schema.ValueKind);
     }
+
+    [Theory]
+    [MemberData(nameof(TestTypes.GetTestDataUsingAllValues), MemberType = typeof(TestTypes))]
+    public static void CreateJsonSchema_ValidateWithTestData(ITestData testData)
+    {
+        // Stress tests the schema generation method using types from the JsonSchemaExporter test battery.
+
+        JsonSerializerOptions options = testData.Options is { } opts
+            ? new(opts) { TypeInfoResolver = TestTypes.TestTypesContext.Default }
+            : TestTypes.TestTypesContext.Default.Options;
+
+        JsonElement schema = AIJsonUtilities.CreateJsonSchema(testData.Type, serializerOptions: options);
+        JsonNode? schemaAsNode = JsonSerializer.SerializeToNode(schema, options);
+
+        Assert.NotNull(schemaAsNode);
+        Assert.Equal(testData.ExpectedJsonSchema.GetValueKind(), schemaAsNode.GetValueKind());
+
+        if (testData.Value is null || testData.WritesNumbersAsStrings)
+        {
+            // Our generated schema does not accept null root values
+            // or numbers formatted as strings, so we skip this test.
```

Comment:
This is on purpose, right? Consider adding "by design" or something similar to the comment, as it currently sounds like it could be a bug.

Reply:
Done.